File: week2 - Arryas/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt user for text
    string text_1 = get_string("Text: ");

    // Count letters
    int letters = count_letters(text_1);

    // Count words
    int words = count_words(text_1);

    // Count sentences
    int sentences = count_sentences(text_1);

    // Display letters, words, sentences
    // printf("Letters %i\n", letters);
    // printf("Words %i\n", words);
    // printf("Sentences %i\n", sentences);

    // Compute The Coleman-Liau index
    float l = letters * 100.0 / words;
    float s = sentences * 100.0 / words;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    // Display Grade
    if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int n = strlen(text);
    int counter = 0;
    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            counter += 1;
        }
    }
    return counter;
}

int count_words(string text)
{
    int n = strlen(text);
    int counter = 0;
    for (int i = 0; i < n; i++)
    {
        if (text[i] == ' ')
        {
            counter += 1;
        }
    }
    counter += 1;
    return counter;
}

int count_sentences(string text)
{
    string marks = "!.?";
    int n = strlen(text);
    int m = strlen(marks);
    int counter = 0;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            if (text[i] == marks[j] && text[i + 1] != '.')
            {
                counter += 1;
            }
        }
    }
    return counter;
}

File: week6 - Python/pset6/readability/readability.py
from cs50 import get_string

text = get_string('Text: ')
text = text.upper()
alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
sentences = ['.', '!', '?']
omit = ['..']
count_letters = 0
count_words = 0
count_sentences = 0
i = 0

for character in text:

    #  Count letters in text
    if character in alphabet:
        count_letters += 1

    #  Count sentences in text
    if character in sentences:
        count_sentences += 1
        # this doesnt consider ..... <- this would be 4 sentences!

    #  Count words in text
    if character == ' ':
        count_words += 1

    i += 1

count_words += 1

# print('Letters:', count_letters, '\nSentences:', count_sentences, '\nWords:', count_words)

#  Count Coleman-Liau index.
# average numbers of letters per 100 words
L = float(count_letters * 100.0 / count_words)

# average number of sentences per 100 words
S = float(count_sentences * 100.0 / count_words)
index = round(0.0588 * L - 0.296 * S - 15.8)

#  Display Grade result
if index <= 1:
    print('Before Grade 1')
elif index >= 16:
    print('Grade 16+')
else:
    print('Grade:', index)